blaze/expr/collections.py

```diff
@@ -100,10 +100,32 @@ def sort(child, key=None, ascending=True):
     ascending : bool, optional
         Determines order of the sort
     """
+    if ascending not in (True, False):
+        msg = "ascending must be True or False, given {}"
+        raise ValueError(msg.format(ascending))
     if not isrecord(child.dshape.measure):
-        key = None
+        if key is None or isinstance(key, _strtypes):
+            # Handle this case separately.
+            return Sort(child, None, ascending)
```

Comment:
When the datashape is array-like and `.sort()` is called with a string value for `key`, I've found no straightforward way to validate that the string is the name of the column being sorted.  We can pass in `key` here instead of `None`, but `Sort()` doesn't do any validation here either.

For example:

```python
In [5]: t
Out[5]: <`t` symbol; dshape='var * {name: string}'>

In [6]: t.name.dshape
Out[6]: dshape("var * string")  # not a record dshape, no `name` column.

In [7]: t.name.sort('name')
Out[7]: t.name.sort(None, ascending=True)

In [8]: t.name.sort()
Out[8]: t.name.sort(None, ascending=True)

In [9]: t.name.sort('foobar')
Out[9]: t.name.sort(None, ascending=True)
```